src/w2grid.js

```diff
@@ -2414,6 +2414,12 @@
                                     this.records.push(data.records[r]);
                                 }
                             }
+                            // columns
+                            if (data.columns) {
+                                for (var r = 0; r < data.columns.length; r++) {
+                                    this.columns.push(data.columns[r]);
+                                }
+                            }
```

Comment:
I am not sure I agree with this. You add the same code to onLoad event and you archived the same thing. Also, what about I click reload button, it will create same columns again.